### PR TITLE
Options

### PR DESCRIPTION
**Pull request by Mantsje**
Adding of new features and settingsObject for the options branch. Further restructuring of the project. Please read ProjectStructure.txt!

### CODE REVIEW

File: sampleExtension/background.js [hunk 1]
```diff
@@ -1,46 +1,57 @@
...
+/* --------------- ------ Updating of variables ------ ---------------*/
 
-// This function adds one url to the blacklist
-addToBlockedSites = function(urlToAdd) {
-    var blockedSiteItem = new BlockedSite(formatUrl(urlToAdd));
-    blockedSites.push(blockedSiteItem);
-    updateStorage();
-    replaceListener();
+addToBlockedSites = function (newUrl, newUrlTitle) {
+    url_formatter.getUrlWithoutServer(newUrl, newUrlTitle, function (url, title) {
```
Comment (rowanvb) [inline sampleExtension/background.js]: TODO extract duplicate function in connectHtmlFunctionality

File: sampleExtension/background.js [hunk 2]
```diff
@@ -51,94 +62,54 @@ addToInterceptDateList = function() {
...
     );
 };
 
+removeWebRequestListener = function() {
+    chrome.webRequest.onBeforeRequest.removeListener(handleInterception);
+};
+
 intercept = function(details) {
-    incrementInterceptionCounter(details.url);
+    console.log("hey intercept");
```
Comment (spazzy) [inline sampleExtension/background.js]: This should probably be removed.

File: sampleExtension/modules/url_formatter.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        url = url_formatter.stripOfScheme(url);
+        url = url_formatter.stripOfFileName(url);
+        callback(url, title);
+    };
+
+}
+
+function Url_Requester() {
+    INVALID_URL_MESSAGE = "We unfortunately could not reach the site you are trying to block.\n" +
+        "Are you sure the url is correct? \n \n";
```
Comment (rowanvb) [inline sampleExtension/modules/url_formatter.js]: TODO: move to constants

File: sampleExtension/modules/url_formatter.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                    callback(xmlHttp.responseURL, title);
+                } else {
+                    callback(xmlHttp.responseURL, theUrlToGet);
+                }
+            } else if (xmlHttp.readyState == 4) {
+                errorHandler(xmlHttp.status);
+            }
+        };
+        xmlHttp.send(null);
+    };
```
Comment (rowanvb) [inline sampleExtension/modules/url_formatter.js]: this function is still quite a mess!

File: sampleExtension/optionsPage/options.html [hunk 5]
```diff
@@ -4,48 +4,70 @@
   <head>
     <title>Options page for The Distraction Shield</title>
     <link rel="stylesheet" type="text/css" href="options.css">
+    <?php include 'urlCheck.php';?>
```
Comment (spazzy) [inline sampleExtension/optionsPage/options.html]: What is this?

File: sampleExtension/optionsPage/options.html [hunk 6]
```diff
@@ -4,48 +4,70 @@
...
+            <input id="turnOff-slider-range" class="range-slider__range" type="range" value="10" min="10" max="10" step="10">
+            for <span class="range-slider__value">00:10 hours.</span>
+            <button id="offBtn">Turn Off</button>
+        </div>
     </div>
 
     <script src="../jquery/jquery-1.10.2.js"></script>
-    <script src="../syncStorage.js"></script>
-    <script src="../blockedSite.js"></script>
+    <script src="../modules/url_formatter.js"></script>
```
Comment (spazzy) [inline sampleExtension/optionsPage/options.html]: Is it necessary to load these "../modules/*"? They are already running in the background(and these can be accessed). Currently the scripts are loaded twice.

File: sampleExtension/Classes/UserSettings.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    };
+
+    this.turnOffFor = function(minutes) {
+        var curDate = new Date();
+        this.status.offTill = new Date(curDate.setMinutes(minutes + curDate.getMinutes()));
+        this.turnOff();
+    };
+
+    this.turnOffForDay = function()  {
+        this.status.offTill = new Date((new Date()).setHours(24,0,0,0));
```
Comment (spazzy) [inline sampleExtension/Classes/UserSettings.js]: Is this necessary?

File: sampleExtension/modules/url_formatter.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    this.getUrlWithoutServer = function(url, title, callback) {
+        url = url_formatter.stripOfScheme(url);
+        url = url_formatter.stripOfFileName(url);
+        callback(url, title);
+    };
+
+}
+
+function Url_Requester() {
+    INVALID_URL_MESSAGE = "We unfortunately could not reach the site you are trying to block.\n" +
```
Comment (spazzy) [inline sampleExtension/modules/url_formatter.js]: Maybe this should be in the constants file

File: sampleExtension/optionsPage/htmlFunctionality.js [hunk 9]
```diff
@@ -54,7 +68,18 @@ initSubmitWithEnter = function(txtFld) {
...
+        synchronizer.syncSettings(settings_object);
     });
 };
+
+
+/* -------------------- Interval slider -------------------- */
+
+initIntervalSlider = function() {
+    intervalSlider = new GreenToRedSlider('#interval-slider');
+    intervalSlider.saveValue = function(value) {
```
Comment (spazzy) [inline sampleExtension/optionsPage/htmlFunctionality.js]: This method is called every "tick" of changing the slider.
A call is made to chrome.sync.storage every call. This makes us exceed the chrome.sync.storage call amount limit. 
So maybe this method should only be called after pressing a "save" button, or only after 2 seconds of no changes.

Comment (Mantsje) [inline sampleExtension/optionsPage/htmlFunctionality.js]: I will look into this for the next iteration.

Comment (spazzy) [inline sampleExtension/optionsPage/htmlFunctionality.js]: Ok. I created an issue for this.

**Pull request merged**